### DISCUSSION TITLE
[Feature] Import JSON config file from the web UI

### DISCUSSION THREAD

**Discussion by truppelito** [immich-app/immich#8908] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Immich has an "Export JSON" button on the Settings page, but no "Import JSON" button.
<img width="368" alt="Screenshot 2024-04-18 at 18 34 01" src="https://github.com/immich-app/immich/assets/1059798/b1f490d8-234d-45de-863a-c8b7382c0846">
I'm aware you can [mount the JSON](https://immich.app/docs/install/config-file/), but a button in the web UI for this would be faster and simpler for many cases, in my opinion. I've personally run into the following two use cases for it:

1) I'm playing around with the settings and just want to go back to a previous configuration
2) I completely deleted my Immich server to start over, and want to restore the settings to what they were (and my docker compose configuration does not mount the JSON file)

Additionally, it may be surprising to people that you can export but not import. I genuinely thought I was missing something in the UI until I went and looked at the docs page above (and asked on Discord).

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
For now you can do this via the API https://immich.app/docs/api/update-config

**Reply by truppelito** [reply]
That's great. Would you consider adding a button to the UI too? I could try my hand at a PR if that's the case.

**Reply by bo0tzz** [reply]
Yes, go ahead.
